session/session.go

```diff
@@ -1590,7 +1590,8 @@ func (s *session) ParseWithParams(ctx context.Context, sql string, args ...inter
 		if digest != nil {
 			// Reset the goroutine label when internal sql execute finish.
 			// Specifically reset in ExecRestrictedStmt function.
-			topsql.AttachSQLInfo(ctx, normalized, digest, "", nil, s.sessionVars.InRestrictedSQL)
+			s.sessionVars.StmtCtx.IsSQLRegistered.Store(true)
+			topsql.AttachSQLInfo(ctx, normalized, digest, s.sessionVars.InRestrictedSQL)
```

Comment:
why is here still a `topsqlstate.TopSQLEnabled()`?

Reply:
No need to attach SQL info here, since later will always attach the SQL and plan information.